gammapy/irf/effective_area_table.py

```diff
@@ -542,6 +545,8 @@ def evaluate(self, offset=None, energy=None):
             val = self._eval_spline(offset.value, np.log10(energy.value))
         else:
             raise ValueError('Invalid interpolation method: {}'.format(method))
+
+        val = np.reshape(val, np.append(offset_shape, energy_shape))
```

Comment:
As it says in the docstring :wink: 

I think shaping stuff should be done in the function that actually calls the interpolator, in this case _eval_linear. Future interpolators might support 2D arrays natively, so the reshaping in the evaluate function would actually be a bad idea